code/iaas/model/src/main/java/io/cattle/platform/core/util/PortSpec.java

```diff
@@ -93,7 +93,7 @@ public Integer getPublicPort() {
         return publicPort;
     }
 
-    public int getPrivatePort() {
+    public Integer getPrivatePort() {
```

Comment:
why this change? private port can never be null. Public port can though

Reply:
then, here https://github.com/carolynhu/cattle/blob/d688ecb308f04ac730cbdd456d751bb1f2663d6f/code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/NetworkDaoImpl.java#L182
should be check "instancePortSpec.getPrivatePort() == servicePortSpec.getPrivatePort()" not equals